test/parallel/test-domain-promise.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const domain = require('domain');
+
+const d = domain.create();
+
+d.run(common.mustCall(() => {
+  Promise.resolve().then(common.mustCall(() => {
+    assert.strictEqual(process.domain, d);
+  }));
+}));
```

Comment:
It would be good to check that domains + promises + vm.runInNewContext() also does the right thing.